cosmos/operators/watcher_kubernetes.py

```diff
@@ -0,0 +1,365 @@
+from typing import TYPE_CHECKING, Any
+
+if TYPE_CHECKING:  # pragma: no cover
+    try:
+        from airflow.sdk.definitions.context import Context
+    except ImportError:
+        from airflow.utils.context import Context  # type: ignore[attr-defined]
+
+try:
+    from airflow.sdk.bases.sensor import BaseSensorOperator
+except ImportError:  # pragma: no cover
+    from airflow.sensors.base import BaseSensorOperator
+from airflow.exceptions import AirflowException
+from airflow.providers.cncf.kubernetes.callbacks import KubernetesPodOperatorCallback, client_type
+
+try:
+    from airflow.providers.standard.operators.empty import EmptyOperator
+except ImportError:  # pragma: no cover
+    from airflow.operators.empty import EmptyOperator  # type: ignore[no-redef]
+from cosmos.airflow._override import CosmosKubernetesPodManager
+from cosmos.config import ProfileConfig
+from cosmos.constants import AIRFLOW_VERSION, PRODUCER_WATCHER_TASK_ID
+from cosmos.operators._watcher.state import build_producer_state_fetcher, get_xcom_val, safe_xcom_push
+from cosmos.operators._watcher.triggerers import WatcherTrigger
+from cosmos.operators.base import (
+    DbtRunMixin,
+    DbtSeedMixin,
+    DbtSnapshotMixin,
+)
+from cosmos.operators.kubernetes import (
+    DbtBuildKubernetesOperator,
+    DbtRunKubernetesOperator,
+    DbtSourceKubernetesOperator,
+)
+
+logger = logging.getLogger(__name__)
+
+CONSUMER_OPERATOR_DEFAULT_PRIORITY_WEIGHT = 10
+PRODUCER_OPERATOR_DEFAULT_PRIORITY_WEIGHT = 9999
+WEIGHT_RULE = "absolute"  # the default "downstream" does not work with dag.test()
+
+
+def _store_dbt_resource_status_from_log(line: str, extra_kwargs: Any) -> None:
+    """
+    Parses a single line from dbt JSON logs and stores node status to Airflow XCom.
+
+    This method parses each log line from dbt when --log-format json is used,
+    extracts node status information, and pushes it to XCom for consumption
+    by downstream watcher sensors.
+    """
+    try:
+        log_line = json.loads(line)
+    except json.JSONDecodeError:
+        logger.info("Failed to parse log: %s", line)
+        log_line = {}
+    node_info = log_line.get("data", {}).get("node_info", {})
+    node_status = node_info.get("node_status")
+    unique_id = node_info.get("unique_id")
+    if unique_id is not None:
+        logger.info("Model: %s is in %s state", unique_id, node_status)
+
+        # TODO: Handle and store all possible node statuses, not just the current success and failed
+        if node_status in ["success", "failed"]:
+            # These lines raise an exception when using KubernetesPodOperatorCallback
+            context = extra_kwargs.get("context")
+            # assert context is not None  # Make MyPy happy
+            safe_xcom_push(task_instance=context["ti"], key=f"{unique_id.replace('.', '__')}_status", value=node_status)
+
+
+producer_task_context = None
+
+
+class WatcherKubernetesCallback(KubernetesPodOperatorCallback):  # type: ignore[misc]
```

Comment:
Create a `_watcher/base.py` with the logic that is common between the `watcher_kubernetes.py` and `watcher.py` modules